src/index.ts

```diff
@@ -42,31 +45,33 @@ async function getTotalVotersBalance(voteData: VotingData): Promise<BigNumber> {
   return totalVotersBalance;
 }
 
-export async function main(): Promise<Map<string, BigNumber>> {
+export async function calculateRewardsAmounts(
+  TotalDistributionAmount: BigNumber
+): Promise<Rewards> {
   const voteData = await fetchVoteData(web3, voteId);
   // snapshot blockNumber of DAO proposal vote
   const blockNumber = voteData.snapshotBlockNumber;
   // fetch total amount of SarcoVR held by voters for a given voteId
   const totalVoteBalance = await getTotalVotersBalance(voteData);
 
-  // construct mapping of voters' addresses and rewards to be allocated
-  const distributionMapping = new Map<string, BigNumber>();
+  // construct object array of voters' addresses and rewards to be allocated
+  const rewardsObject: Rewards = [];
   for (let i = 0; i < voteData.addresses.length; i++) {
     const votingAddress = voteData.addresses[i];
     const stakedValueAt: BigNumber = await stakingContract.methods
       .stakeValueAt(votingAddress, blockNumber)
       .call();
-
     // percentage and distributionAmount calculations include 'factor' to enable a good decimal approximation in the computation
+    const factor = TotalDistributionAmount.div(100000);
     const percentage = BigNumber.from(stakedValueAt).mul(factor).div(totalVoteBalance);
-    const distributionAmount = DISTRIBUTION_AMOUNT.div(factor).mul(percentage);
+    const distributionAmount = TotalDistributionAmount.div(factor).mul(percentage);
 
-    distributionMapping.set(votingAddress, distributionAmount);
+    rewardsObject[i] = { _address: votingAddress, _amount: distributionAmount };
   }
 
-  return distributionMapping;
+  return rewardsObject;
 }
 
-(async () => {
-  await main();
-})();
+// (async () => {
+//   await calculateRewardsAmounts();
+// })();
```

Comment:
Does this need to be removed?

Reply:
Uncommented and added a console.log, as well as an env variable to simulate the distributions

<img width="844" alt="Screenshot 2022-10-21 at 09 56 16" src="https://user-images.githubusercontent.com/105349716/197143812-9bff092e-5cc3-4cc0-91e1-3ca4c617cf60.png">